### ISSUE TITLE
NoClassDefFoundError: me/realized/tm/api/TMAPI

### ISSUE THREAD

**Issue by mibby** [Realizedd/TokenManager#10]
Since updating to TokenManager v3.x, I've been getting these errors when using in conjunction with the plugin BossShopPro and the module ThirdCurrency to use TokenManager currency in Boss shops. These errors do not occur in TokenManager 2.x.

Did something break or get changed in token?

https://paste.ubuntu.com/p/6CPZkSXgWr/
```
[11:15:44] [Server thread/ERROR]: Could not pass event BSTransformStringEvent to ThirdCurrency v1.1.4
java.lang.NoClassDefFoundError: me/realized/tm/api/TMAPI
	at org.black_ixx.bossshop.pointsystem.BSPointsPluginTokenManager.getPoints(BSPointsPluginTokenManager.java:19) ~[BossShopPro.jar:?]
	at org.black_ixx.bossshop.managers.features.PointsManager.getPoints(PointsManager.java:173) ~[BossShopPro.jar:?]
	at org.black_ixx.bossshop.addon.thirdcurrency.BSListener.transformString(BSListener.java:35) ~[BSP-ThirdCurrency.jar:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor1304.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:70) ~[patched_1.12.2.jar:git-Paper-1386]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:78) ~[patched_1.12.2.jar:git-Paper-1386]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched_1.12.2.jar:git-Paper-1386]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:515) ~[patched_1.12.2.jar:git-Paper-1386]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:500) ~[patched_1.12.2.jar:git-Paper-1386]
	at org.black_ixx.bossshop.managers.misc.StringManager.transform(StringManager.java:90) ~[BossShopPro.jar:?]
	at org.black_ixx.bossshop.core.rewards.BSRewardTypeCommand.giveReward(BSRewardTypeCommand.java:43) ~[BossShopPro.jar:?]
	at org.black_ixx.bossshop.core.BSBuy.purchase(BSBuy.java:522) ~[BossShopPro.jar:?]
	at org.black_ixx.bossshop.core.BSBuy.purchaseTask(BSBuy.java:465) ~[BossShopPro.jar:?]
	at org.black_ixx.bossshop.core.BSBuy.purchaseTry(BSBuy.java:440) ~[BossShopPro.jar:?]
	at org.black_ixx.bossshop.core.BSBuy.click(BSBuy.java:410) ~[BossShopPro.jar:?]
	at org.black_ixx.bossshop.listeners.InventoryListener.purchase(InventoryListener.java:197) ~[BossShopPro.jar:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor1269.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:70) ~[patched_1.12.2.jar:git-Paper-1386]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:78) ~[patched_1.12.2.jar:git-Paper-1386]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched_1.12.2.jar:git-Paper-1386]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:515) ~[patched_1.12.2.jar:git-Paper-1386]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:500) ~[patched_1.12.2.jar:git-Paper-1386]
	at net.minecraft.server.v1_12_R1.PlayerConnection.a(PlayerConnection.java:1981) ~[patched_1.12.2.jar:git-Paper-1386]
	at net.minecraft.server.v1_12_R1.PacketPlayInWindowClick.a(SourceFile:33) ~[patched_1.12.2.jar:git-Paper-1386]
	at net.minecraft.server.v1_12_R1.PacketPlayInWindowClick.a(SourceFile:10) ~[patched_1.12.2.jar:git-Paper-1386]
	at net.minecraft.server.v1_12_R1.PlayerConnectionUtils.lambda$ensureMainThread$0(PlayerConnectionUtils.java:14) ~[patched_1.12.2.jar:git-Paper-1386]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_162]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_162]
	at net.minecraft.server.v1_12_R1.SystemUtils.a(SourceFile:46) [patched_1.12.2.jar:git-Paper-1386]
	at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:843) [patched_1.12.2.jar:git-Paper-1386]
	at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:426) [patched_1.12.2.jar:git-Paper-1386]
	at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:767) [patched_1.12.2.jar:git-Paper-1386]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:665) [patched_1.12.2.jar:git-Paper-1386]
```

**Comment by Realizedd** [comment]
I've rewritten the whole project in v3, so I must've forgotten to add back the old API and mark it as deprecated. Will do so in the next version.

**Comment by mibby** [comment]
Thanks. 

Is there any way to still use the old method in the BossShop module? As far as I know, it doesn't hook into any specific plugin api, but rather you can specify _any_ plugin to use as the third currency.

This is how it looks;

```
  PointsPlugin: TokenManager
```

**Comment by Realizedd** [comment]
Well, by the looks of the stacktrace, it does hook into the old API. So there is probably no other way than switching back to the old version until I finish the update.

Edit: You could also request the author of BossShopPro to update the TokenManager hook to v3.

**Comment by Realizedd** [comment]
Should be fixed in 1e9aeb6. Let me know if it isn't.
